frontend/src/components/v1/chat/event-message-components/finish-event-message.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { ActionEvent } from "#/types/v1/core";
+import { ChatMessage } from "../../../features/chat/chat-message";
+import { MicroagentStatusWrapper } from "../../../features/chat/event-message-components/microagent-status-wrapper";
+// TODO: Implement V1 LikertScaleWrapper when API supports V1 event IDs
+// import { LikertScaleWrapper } from "../../../features/chat/event-message-components/likert-scale-wrapper";
+import { getEventContent } from "../event-content-helpers/get-event-content";
+import { MicroagentStatus } from "#/types/microagent-status";
+
+interface FinishEventMessageProps {
+  event: ActionEvent;
```

Comment:
```suggestion
  event: ActionEvent<FinishAction>;
```

Don’t forget to pass generics here as well. That way we ensure the parent component passes the right type, and we don’t have to include conditionals like:

```ts
if (event.action.kind !== "FinishAction") {
  return null;
}
```